### DISCUSSION TITLE
How to tell if the app has been minimized or moved to the background

### DISCUSSION THREAD

**Discussion by DarrylDreiling-Sphero** [microsoft/WindowsAppSDK#4357] [General] [answered]
Our application allows kids to interact with our robots.  We would like to be able to stop the robot if the app is minimized or is no longer in the forefront.  I have not been able to find a way to detect this.  Hoping someone has some advice on how to handle this.

**Comment by castorix** [comment, marked as answer]
On Windows 10, when it is minimized, it is not in Foreground anymore
So you can just test if it loses Foreground state, for example with _Activated_ :

` this.Activated += MainWindow_Activated;`

with :

        private void MainWindow_Activated(object sender, WindowActivatedEventArgs args)
        {
            if (args.WindowActivationState == WindowActivationState.Deactivated)
                Console.Beep(8000, 10);
        }

**Reply by DarrylDreiling-Sphero** [reply]
Thank you so much for the help!  This is working well for me since I only want to know when the app is "Deactivated".  This does seem a little strange to me that there is no "Activated" state.  When I tested with with " CodeActivated" it did not work as I expected it to.  For my however, this is working exactly I need.

Thanks!
